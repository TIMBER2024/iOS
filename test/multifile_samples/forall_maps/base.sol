pragma solidity 0.8.4;

contract Base {
    // Simple map
    mapping(uint => uint) a;

    // Map with complex key
    mapping (string => int16) public c;

    string sS;

    // Nested map
    mapping (string => mapping(uint8 => int8)) d;

    // Map to array forall
    mapping (uint => uint[]) e;

    // Array of maps, array forall only
    mapping (uint => uint)[] f;

    // Array of maps, map forall only
    mapping (uint => uint)[] g;

    // Nested map first map only
    mapping (string => mapping(uint8 => int8)) h;

    // Nested map last map only
    mapping (string => mapping(uint8 => int8)) i;

    struct SA {
        mapping(string=>uint) m;
        uint cnt;
    }

    struct SB {
        SA[] sas;
    }

    SB j;
}
