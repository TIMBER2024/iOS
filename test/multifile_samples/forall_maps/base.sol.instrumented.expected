pragma solidity 0.8.4;

import "./__scribble_ReentrancyUtils.sol";

contract Base {
    struct SA {
        string_to_uint256.S m;
        uint cnt;
    }

    struct SB {
        SA[] sas;
    }

    uint256_to_uint256.S internal a;
    string_to_int16.S c0;
    string internal sS;
    string_to_uint8_to_int8_S_471.S internal d;
    uint256_to_uint256_arr.S internal e;
    mapping(uint => uint)[] internal f;
    uint256_to_uint256.S[] internal g;
    string_to_mapping_uint8_to_int8.S internal h;
    mapping(string => uint8_to_int8.S) internal i;
    SB internal j;

    function c(string memory ARG_0) public returns (int16 RET_0) {
        return string_to_int16.get(c0, ARG_0);
    }
}